src/decoder/serde.rs

```diff
@@ -701,6 +701,18 @@ impl<'de> Deserialize<'de> for RegExp {
     }
 }
 
+impl<'de> Deserialize<'de> for JavaScriptCodeWithScope {
```

Comment:
This follows the pattern of the other deserializers defined here. Since the deserialize implementation for Bson implements deserializing from extjson, we piggyback off that to either get a JavaScriptCodeWithScope or return an error.